stacks-signer/src/utils.rs

```diff
@@ -138,3 +143,18 @@ pub fn build_stackerdb_contract(
     stackerdb_contract += "    ";
     stackerdb_contract
 }
+
+/// Helper function to convert a private key to a Stacks address
+pub fn to_addr(stacks_private_key: &StacksPrivateKey, network: &Network) -> StacksAddress {
+    let version = match network {
+        Network::Mainnet => C32_ADDRESS_VERSION_MAINNET_SINGLESIG,
+        Network::Testnet | Network::Mocknet => C32_ADDRESS_VERSION_TESTNET_SINGLESIG,
+    };
+    StacksAddress::from_public_keys(
+        version,
+        &AddressHashMode::SerializeP2PKH,
+        1,
+        &vec![StacksPublicKey::from_private(stacks_private_key)],
+    )
+    .expect("BUG: failed to generate address from private key")
+}
```

Comment:
I'd suggest using a `mainnet` bool and the `p2pkh()` method in `StacksAddress`:

```suggestion
pub fn to_addr(stacks_private_key: &StacksPrivateKey, network: &Network) -> StacksAddress {
    // this could be a function of Network, so you would call `network.is_mainnet()`
    let is_mainnet = match network {
        Network::Mainnet => true,
        Network::Testnet | Network::Mocknet => false,
    };
    StacksAddress::p2pkh(is_mainnet, &StacksPublicKey::from_private(stacks_private_key))
}
```

That way, we avoid a `vec![]`, an `.expect()`, and also this code doesn't have to worry about getting the version flags right.